src/game/savegame.c

```diff
@@ -492,6 +512,17 @@ void Savegame_ScanSavedGames(void)
             }
             g_SavedGamesCount++;
         }
+
+        // Calculate savegame features.
+        if (savegame_info->format != SAVEGAME_FORMAT_LEGACY) {
+            savegame_info->restart = true;
+            savegame_info->select_level =
+                g_GameInfo.save_initial_version >= SAVEGAME_VERSION_1 ? true
+                                                                      : false;
+        } else {
+            savegame_info->restart = false;
+            savegame_info->select_level = false;
+        }
```

Comment:
These shouldn't be filled by the SG scanner but by the loaders themselves at earlier stage. Please move these assignments to relevant places, for example like this:

```diff
diff --git a/src/game/savegame.c b/src/game/savegame.c
index c1cb1784..0d2fd3db 100644
--- a/src/game/savegame.c
+++ b/src/game/savegame.c
@@ -512,17 +512,6 @@ void Savegame_ScanSavedGames(void)
             }
             g_SavedGamesCount++;
         }
-
-        // Calculate savegame features.
-        if (savegame_info->format != SAVEGAME_FORMAT_LEGACY) {
-            savegame_info->restart = true;
-            savegame_info->select_level =
-                g_GameInfo.save_initial_version >= SAVEGAME_VERSION_1 ? true
-                                                                      : false;
-        } else {
-            savegame_info->restart = false;
-            savegame_info->select_level = false;
-        }
     }
 
     REQUEST_INFO *req = &g_SavegameRequester;
diff --git a/src/game/savegame_bson.c b/src/game/savegame_bson.c
index bcd1bd7e..ce20f924 100644
--- a/src/game/savegame_bson.c
+++ b/src/game/savegame_bson.c
@@ -1024,6 +1024,13 @@ bool Savegame_BSON_FillInfo(MYFILE *fp, SAVEGAME_INFO *info)
         ret = info->level_num != -1;
     }
     json_value_free(root);
+
+    SAVEGAME_BSON_HEADER header;
+    File_Seek(fp, 0, FILE_SEEK_SET);
+    File_Read(&header, sizeof(SAVEGAME_BSON_HEADER), 1, fp);
+    info->restart = true;
+    info->select_level = header.initial_version >= SAVEGAME_VERSION_1;
+
     return ret;
 }
 
diff --git a/src/game/savegame_legacy.c b/src/game/savegame_legacy.c
index c0e697c4..7eb615c3 100644
--- a/src/game/savegame_legacy.c
+++ b/src/game/savegame_legacy.c
@@ -405,6 +405,9 @@ bool Savegame_Legacy_FillInfo(MYFILE *fp, SAVEGAME_INFO *info)
     File_Read(&level_num, sizeof(int16_t), 1, fp);
     info->level_num = level_num;
 
+    info->restart = false;
+    info->select_level = false;
+
     return true;
 }
 
```